esx_service/utils/auth_data.py

```diff
@@ -576,7 +638,9 @@ def connect(self):
 
         if not need_create_table:
             if os.path.isfile(self.db_path) and self.need_upgrade_db():
-                self.handle_upgrade_db()
+                error = self.handle_upgrade_db()
+                if error:
```

Comment:
Since we open , use and close the DB for each request from Docker then no, we should not check at the start only, it does not really improve anything

if the upgrade fails, there is no data loss. Its a corner case (i.e. very rare, preferably never) and the workaround is to remove the DB. I do not think we should carry forward almost-useless-code reading both versions (and writing both versions), and then test it too.